hudi-hadoop-mr/src/main/java/org/apache/hudi/hadoop/InputPathHandler.java

```diff
@@ -95,19 +101,20 @@ private void parseInputPaths(Path[] inputPaths, List<String> incrementalTables)
           // We already know the base path for this inputPath.
           basePathKnown = true;
           // Check if this is for a snapshot query
-          String tableName = metaClient.getTableConfig().getTableName();
-          tagAsIncrementalOrSnapshot(inputPath, tableName, metaClient, incrementalTables);
+          tagAsIncrementalOrSnapshot(inputPath, metaClient, incrementalTables);
           break;
         }
       }
       if (!basePathKnown) {
-        // This path is for a table that we dont know about yet.
+        // This path is for a table that we don't know about yet.
         HoodieTableMetaClient metaClient;
         try {
           metaClient = getTableMetaClientForBasePath(inputPath.getFileSystem(conf), inputPath);
+          String databaseName = metaClient.getTableConfig().getDatabaseName();
           String tableName = metaClient.getTableConfig().getTableName();
-          tableMetaClientMap.put(tableName, metaClient);
-          tagAsIncrementalOrSnapshot(inputPath, tableName, metaClient, incrementalTables);
+          tableMetaClientMap.put(isIncrementalUseDatabase && !StringUtils.isNullOrEmpty(databaseName)
+                  ? databaseName + "." + tableName : tableName, metaClient);
```

Comment:
can we move the table name creation logic into a helper method?

Reply:
Modified

Reply:
The global variable isincrementalusedatabase is currently only used by assistant methods. Do I need to remove it? Or keep it for later use in other places